### PR TITLE
Add support for releasing the extension out of band with MSAL

### PR DESCRIPTION
**Pull request by bgavrilMS**

### CODE REVIEW

File: build/pipeline-releasebuild.yaml [hunk 1]
```diff
@@ -39,6 +39,9 @@ steps:
     BuildPlatform: '$(BuildPlatform)'
     BuildConfiguration: '$(BuildConfiguration)'
     MsalClientSemVer: $(MsalClientSemVer)
+    # Leave empty to release the desktop extenion with a project refrence to msal
```
Comment (jmprieur) [inline build/pipeline-releasebuild.yaml]: extenion => extension

File: build/pipeline-releasebuild.yaml [hunk 2]
```diff
@@ -39,6 +39,9 @@ steps:
...
+    # Or set a version to take a dependency on a pre-released MSAL NuGet package
```
Comment (jmprieur) [inline build/pipeline-releasebuild.yaml]: pre-released? do you mean previously released ?

File: src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj [hunk 3]
```diff
@@ -19,17 +20,22 @@
...
-    <MsalClientSemVer Condition="'$(MsalClientSemVer)' != ''">$(MsalClientSemVer)-preview</MsalClientSemVer>
-    <!--This will generate AssemblyVersion, AssemblyFileVersion and AssemblyInformationVersion-->
-    <Version>$(MsalClientSemVer)</Version>
+    <DesktopExtensionVersion Condition="'$(ExtensionUsesMsalFromNuGetVersion)' == '' and '$(MsalClientSemVer)' != ''">
+      $(MsalClientSemVer)-preview
+    </DesktopExtensionVersion>
+    <DesktopExtensionVersion  Condition="'$(ExtensionUsesMsalFromNuGetVersion)' != ''" >
+      $(ExtensionUsesMsalFromNuGetVersion)-preview
+    </DesktopExtensionVersion>
+    <DesktopExtensionVersion Condition="$(DesktopExtensionVersion) == ''">4.5.0-internal</DesktopExtensionVersion>
```
Comment (jmprieur) [inline src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj]: Is it something that is hardcoded?

File: src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj [hunk 4]
```diff
@@ -82,8 +88,12 @@
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta2-18618-05" PrivateAssets="All" />
   </ItemGroup>
 
-  <ItemGroup>
+  <!-- Set the variable MsalPackageReferenceVersion to an MSAL version to relesae the extension out of band with MSAL-->
```
Comment (jennyf19) [inline src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj]: Relesae -> release

File: src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj (hunk 3 above)
Comment (bgavrilMS) [reply src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj]: Yes, but I will remove this as there are so many defaults now ...

File: src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj [hunk 5]
```diff
@@ -3,17 +3,16 @@
   <PropertyGroup>
     <TargetFrameworkNetDesktop>net45</TargetFrameworkNetDesktop>
     <TargetFrameworkNetStandard>netstandard2.0</TargetFrameworkNetStandard>
-    <TargetFrameworkNetCore2>netcoreapp2.1</TargetFrameworkNetCore2>
-    <TargetFrameworkNetCore3>netcoreapp3.0</TargetFrameworkNetCore3>
```
Comment (bgavrilMS) [inline src/extensions.msal/Microsoft.Identity.Client.Extensions.Msal/Microsoft.Identity.Client.Extensions.Msal.csproj]: Looks like the extension was supposed to be built for .net core 2 and .net core 3, but there is no .net core 3 logic right now, so reverting this. Also, the flags were wrong. We need a lot of scrutiny for releasing the extension from here ...

**Comment by bgavrilMS** [comment]
After further inspection I've decided to drop this. The ADAL + MSAL desktop extensions share a lot of code between them and there are many unanswered questions related to shipping in sync with ADAL / MSAL and out of band. The ADAL extension has not been ported to ADAL, and clearly it will be updated much more frequently than ADAL itself. 

If we are going to add all this functionality into the libraries, then we need to find an answer to the shared functionality problem. Until then, the existing repo https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet and out of band releases are more tractable. 

CC @henrik-me

**Pull request closed without merging**